### ISSUE TITLE
{path} placeholders are URLEncoded making usage of composite url paths harder.

### ISSUE THREAD

**Issue by dpopov76** [canton7/RestEase#24]
If I have the following interface:
```
public interface ITest
{
  [Post("/v1/Storage/{path}")]
  Task Upload([Path] string path, [Body] Stream data);
}
```

and I call it like that:
  ```
var api = RestClient.For<ITest>("http://localhost");
  var data = new Stream(...);
  api.Upload("volumeA/partitionB/test.txt", data);
```

it fails because the path is URLEncoded in `Requester.SubstitutePathParameters(...)` and the resulting request looks like this:

POST http://localhost/v1/Storage/volumeA%2FpartitionB%2Ftest.txt
while my service endpoint expects:
POST http://localhost/v1/Storage/volumeA/partitionB/test.txt

The only workaround I found is to declare the interface like that:

 ```
 [Post("/v1/Storage/{volume}/{partition}/{filename}")]
  Task Upload([Path] string volume, [Path] string partition, [Path] string filename, [Body] Stream data);
```

and call it like that, which reduces the flexibility of the method:
` api.Upload("volumeA", "partitionB", "test.txt", data);`

I understand the reason behind URLEncode, but it will be nice if paths could be handled a little bit more precisely. One option is to change the Requester.SubstitutePathParameters() so UrlEncode is used like this:
```
serialized.Value = string.Join('/', serialized.Value.Split('/').Select(s => UrlEncode(s)));
```

i.e. the individual "folders" in the path parameter are URLEncoded, but the existing path separators (backsplashes) are preserved.

**Comment by canton7** [comment]
Hmm, I used to allow '/'s in path parameters, but then I [changed my mind](https://github.com/canton7/RestEase/commit/d0dd8ba713bb2b0ed29b470710d471bf833980c9). However, in doing so, I broke one of the FAQ items.

I think the best thing to do is probably add a property to PathAttribute allowing URL escaping to be turned off? That way we don't break backwards compatibility and the use-case of inserting raw URL segments can be supported in full, rather than being a weird mis-match between escaping most things, but not /'s...
